src/Contracts/AttributeEncoder.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * This file is part of the Laravel Auditing package.
+ *
+ * @author     Antério Vieira <anteriovieira@gmail.com>
+ * @author     Quetzy Garcia  <quetzyg@altek.org>
+ * @author     Raphael França <raphaelfrancabsb@gmail.com>
+ * @copyright  2015-2018
+ *
+ * For the full copyright and license information,
+ * please view the LICENSE.md file that was distributed
+ * with this source code.
+ */
+
+namespace OwenIt\Auditing\Contracts;
+
+interface AttributeEncoder extends AttributeModifier
+{
+    /**
+     * Encode an attribute value.
+     *
+     * @param mixed $value
+     *
+     * @return string
+     */
+    public static function encode($value): string;
+
+    /**
+     * Decode an attribute value.
+     *
+     * @param mixed $value
+     *
+     * @return string
+     */
+    public static function decode($value): string;
```

Comment:
@quetzyg Any reason why the encoded/decoded value must be a `string`? A custom encoder could also return `null` here.

My use case (the auditable model has binary IP address attributes that may be null):
```php
    public static function encode($value)
    {
        return is_null($value) ? null : inet_ntop($value);
    }

    public static function decode($value)
    {
        return $value ? inet_pton($value) : null;
    }
```

Reply:
Good point! I'll update both methods so that there's no return type enforced.

Reply:
[Updated](https://github.com/owen-it/laravel-auditing/commit/277f1c1f151ba7a77d5e967098f4fe2d005dfa71)